src/sentry/api/endpoints/project_triage_stats.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from datetime import timedelta
+from django.utils import timezone
+
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.models import Group, GroupStatus, Activity
+from sentry.utils.dates import to_timestamp
+
+
+class TriageStatus(GroupStatus):
+    DNE = -1
+    ASSIGNED = 6
+
+
+class ProjectTriageStatsEndpoint(ProjectEndpoint):
+
+    def filter_activity_set(self, set, cutoff):
+        return [ai for ai in set if ai.datetime < cutoff]
+
+    def process_activity_set(self, group, dates):
+
+        activity_set = Activity.objects.filter(
+            group=group.id).order_by('datetime')
+
+        status = GroupStatus.UNRESOLVED
+        statuses = []
+        for date in dates:
+            ts = int(to_timestamp(date))
+            if date < group.first_seen:
+                statuses.append({'timestamp': ts, 'status': TriageStatus.DNE})
+                continue
+            sliced = self.filter_activity_set(activity_set, date)
+
+            for activity in sliced:
+                if activity.type in set([Activity.SET_RESOLVED, Activity.SET_RESOLVED_IN_RELEASE,
+                                         Activity.SET_RESOLVED_BY_AGE, Activity.SET_RESOLVED_IN_COMMIT]):
+                    status = GroupStatus.RESOLVED
+
+                elif activity.type in set([Activity.SET_UNRESOLVED, Activity.SET_REGRESSION]):
+                    status = GroupStatus.UNRESOLVED
+
+                elif activity.type is Activity.SET_IGNORED:
+                    status = GroupStatus.IGNORED
+
+                elif activity.type is Activity.ASSIGNED:
+                    if status == GroupStatus.UNRESOLVED:
+                        status = TriageStatus.ASSIGNED
+
+                elif activity.type is Activity.UNASSIGNED:
+                    if status == TriageStatus.ASSIGNED:
+                        prev_statuses = filter(
+                            lambda s: s['status'] != TriageStatus.ASSIGNED, statuses)
+                        if len(prev_statuses) == 0:
+                            prev_statuses = [
+                                {'status': GroupStatus.UNRESOLVED}]
+                        status = prev_statuses[-1]['status']
+
+                elif activity.type in set([Activity.MERGE, Activity.NOTE]):
+                    pass
+                else:
+                    pass
+
+            statuses.append({'timestamp': ts, 'status': status})
+
+        return statuses
+
+    def get(self, request, project):
+
+        now = timezone.now()
+        dates = [now - timedelta(days=i) for i in range(30)]
+
+        groups = Group.objects.filter(
+            project=project.id,
+            last_seen__gte=now - timedelta(days=30)
+        )
```

Comment:
The only real suggestion I'm going to make here since this may be rather large, you don't need the full Group objects in memory. So you can just `.values_list('id', 'first_seen')` then work with the 2-tuples.